### PR TITLE
Gk tidy

### PR DESCRIPTION
**Pull request by gilesknap**
Tidy up code:

- standardize imports
- pull init parameter types into local namespace for access by subclasses
- move hook registration to setup()
- declare attributes and methods at the class level when __init__() not required

### CODE REVIEW

File: malcolm/core/__init__.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
 from .notifier import Notifier
 from .part import Part, PartRegistrar, APartName
 from .process import Process, ProcessPublishHook, ProcessStartHook, \
-    ProcessStopHook, APublished, UnpublishedInfo, UUnpublishedInfos
+    ProcessStopHook, APublished, UnpublishedInfo, UUnpublishedInfos, \
+    AHookable
```
Comment (thomascobb) [inline malcolm/core/__init__.py]: AHookable should come from .hook

File: malcolm/core/process.py [hunk 2]
```diff
@@ -28,6 +28,9 @@
 with Anno("The list of currently published Controller mris"):
     APublished = Array[str]
 
+# Pull re-used annotypes into our namespace in case we are subclassed
+AHookable = AHookable
```
Comment (thomascobb) [inline malcolm/core/process.py]: Don't need this as it should be exported from .hook in core/__init__.py

File: malcolm/modules/ADCore/parts/detectordriverpart.py [hunk 3]
```diff
@@ -23,11 +20,19 @@
 with Anno("Directory to write data to"):
     AFileDir = str
 
+# Pull re-used annotypes into our namespace in case we are subclassed
+APartName = APartName
+AMri = builtin.parts.AMri
+AbortHook = scanning.hooks.AbortHook
```
Comment (thomascobb) [inline malcolm/modules/ADCore/parts/detectordriverpart.py]: I think we shouldn't pull hook annotypes into our namespace, just __init__ types

File: malcolm/modules/ADCore/parts/detectordriverpart.py [hunk 4]
```diff
@@ -141,7 +151,7 @@ def configure(self,
                   completed_steps,  # type: ACompletedSteps
                   steps_to_do,  # type: AStepsToDo
                   part_info,  # type: APartInfo
-                  generator,  # type: AGenerator
+                  generator,  # type: scanning.util.AGenerator
```
Comment (thomascobb) [inline malcolm/modules/ADCore/parts/detectordriverpart.py]: I reckon all the other hook annotypes should be namespaced too

File: malcolm/modules/ADCore/parts/positionlabellerpart.py [hunk 5]
```diff
@@ -18,37 +18,42 @@
 if TYPE_CHECKING:
     from typing import Tuple
 
+# Pull re-used annotypes into our namespace in case we are subclassed
```
Comment (thomascobb) [inline malcolm/modules/ADCore/parts/positionlabellerpart.py]: Not needed as we don't override __init__

Comment (thomascobb) [reply malcolm/modules/ADCore/parts/positionlabellerpart.py]: Although this does violate the "all annotypes should be there if we are subclassed" principle, but it's probably clearer to only pull annotypes into our namespace if we are using them

File: malcolm/modules/ADPandABlocks/parts/pandadatasetbussespart.py [hunk 6]
```diff
@@ -1,25 +1,24 @@
 from annotypes import TYPE_CHECKING
 
 from malcolm.core import PartRegistrar
-from malcolm.modules.pandablocks.parts.pandabussespart import PandABussesPart
-from malcolm.modules.pandablocks.util import PositionCapture
+from malcolm.modules import pandablocks
```
Comment (thomascobb) [inline malcolm/modules/ADPandABlocks/parts/pandadatasetbussespart.py]: Can be merged with the next line

File: malcolm/modules/pandablocks/controllers/pandablockcontroller.py [hunk 7]
```diff
@@ -12,7 +12,7 @@
 from ..parts.pandaactionpart import PandAActionPart
 from ..parts.pandafieldpart import PandAFieldPart
 from ..parts.pandatablepart import PandATablePart
-from ..util import AClient, ADocUrlBase, SVG_DIR, ABlockName
+from .. import util
```
Comment (thomascobb) [inline malcolm/modules/pandablocks/controllers/pandablockcontroller.py]: Why not from ..util import AClient, etc ?

File: malcolm/modules/pandablocks/parts/__init__.py [hunk 8]
```diff
@@ -1 +1,3 @@
-# No imports as nothing should be instantiated from YAML
+# nothing should be instantiated from YAML but pandabussespart is imported
+# from ADPandaBlocks
+from . import pandabussespart
```
Comment (thomascobb) [inline malcolm/modules/pandablocks/parts/__init__.py]: Why not
```from .pandabussespart import ...```

File: malcolm/modules/ADCore/parts/detectordriverpart.py (hunk 4 above)
Comment (gilesknap) [reply malcolm/modules/ADCore/parts/detectordriverpart.py]: Sorry, I don't understand this comment.

Comment (gilesknap) [reply malcolm/modules/ADCore/parts/detectordriverpart.py]: got it.

File: malcolm/modules/ADCore/parts/detectordriverpart.py [hunk 9]
```diff
@@ -137,11 +142,11 @@ def configure_args_with_exposure(self, keys):
...
-                  context,  # type: AContext
-                  completed_steps,  # type: ACompletedSteps
-                  steps_to_do,  # type: AStepsToDo
-                  part_info,  # type: APartInfo
-                  generator,  # type: AGenerator
+                  context,  # type: scanning.hooks.AContext
+                  completed_steps,  # type: scanning.hooks.ACompletedSteps
+                  steps_to_do,  # type: scanning.hooks.AStepsToDo
+                  part_info,  # type: scanning.hooks.APartInfo
+                  generator,  # type: scanning.util.AGenerator
```
Comment (thomascobb) [inline malcolm/modules/ADCore/parts/detectordriverpart.py]: scanning.hooks?

File: malcolm/modules/ADCore/parts/detectordriverpart.py [hunk 10]
```diff
@@ -137,11 +142,11 @@ def configure_args_with_exposure(self, keys):
...
                   fileDir,  # type: AFileDir
```
Comment (thomascobb) [inline malcolm/modules/ADCore/parts/detectordriverpart.py]: And isn't this in scanning.hooks too?

**Comment by gilesknap** [comment]
I moved all hook function annotypes to scanning.hooks.
I had to move DatasetType from scanning.util to scanning.infos to avoid a circular import.
Presently other consumers of DatasetType still import it from scanning.util.

Let me know if you have a nicer solution.

**Pull request merged**